### PR TITLE
Adding file_lab.py

### PR DESCRIPTION
**Pull request by lambocalrissian**
I really liked using the pathlib library. Easier to read IMO

### CODE REVIEW

**Comment by lambocalrissian** [comment]
Added mailroom2 assignment. Pathlib came in handy again. I'm really liking Pathlib

File: students/Gabe_Lamberth/lesson04/file_lab.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            continue
+        else:
+            print(f)
+
+
+def copy_file():
+    """Copy a file from a source to a destination (~1 min for a 3GB file)."""
+    file_list()
+    source = pathlib.Path(input("\nEnter the file to copy: "))
+    """Sending file to users home directory"""
```
Comment (mdgregor) [inline students/Gabe_Lamberth/lesson04/file_lab.py]: for single line comments use `#` instead

File: students/Gabe_Lamberth/lesson04/file_lab.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    """Found this solution from another student, but added slight modification"""
+    with source.open('rb') as source_file, destination.open('wb') as dest_file:
+        source_size = os.path.getsize(str(source))
+        chunk_size = 1000000
+        num_steps = source_size // chunk_size
+        print("Copying...")
+        """This for loop iterates without using the counter"""
+        for _ in range(num_steps):
+            dest_file.write(source_file.read(chunk_size))
+        dest_file.write(source_file.read())
```
Comment (mdgregor) [inline students/Gabe_Lamberth/lesson04/file_lab.py]: yeah we use this kind of pattern in the networking class doing things in chunks

File: students/Gabe_Lamberth/lesson04/file_lab.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+def main():
+    print('\n>>>>>>>>>>Welcome to the Python File Program<<<<<<<<<<')
+    print('The current time and date is:', time.asctime())
+    user_prompt = ("\nWhat would you like to do?\n"
+                   "Type '1- To Display all contents in current directory\n"
+                   "Type '2- To Read a file'\n"
+                   "Type '3- To Copy a file\n"
+                   "Type 'q- To end the program'\n"
+                   "Enter value here >> ")
```
Comment (mdgregor) [inline students/Gabe_Lamberth/lesson04/file_lab.py]: you could probably make this easier by just doing triple `"""`

File: students/Gabe_Lamberth/lesson04/mailroom2.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            patrons[choice].append(float(amount))
+            send_email(choice, float(amount))
+        else:
+            selection = input(f'{choice} already found in database, add another donation? \n Enter Yes or No>> ')
+            if selection.lower() == 'yes':
+                amount = input('Please enter a donation amount: ')
+                patrons[choice].append(float(amount))
+                send_email(choice, float(amount))
+            else:
+                continue
```
Comment (mdgregor) [inline students/Gabe_Lamberth/lesson04/mailroom2.py]: no need for the `else: continue` here,  it will go to the top of the `while` loop anyway

**Pull request merged**